### PR TITLE
Cannot Add OST

### PR DESCRIPTION
**Pull request by johnsonw**
Provide documentation on analysis and solution for not being able to add
an ost.

Signed-off-by: Will Johnson <william.c.johnson@intel.com>

### CODE REVIEW

File: docs/support/cannot-add-osts.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+uuid: 05fa54cd-7d12-4090-9c8f-53441f158d12:
+{"target_name": "home-OST0001", "bdev": "/dev/mapper/360060e8016641330000166130000c021", "backfstype": "ext4", "mntpt": "/mnt/home-OST0001"}
+...
+```
+
+Update the `bdev` property with the alias path that multipath has defined and save the file:
+```
+{"target_name": "home-OST0001", "bdev": "/dev/mapper/ost01", "backfstype": "ext4", "mntpt": "/mnt/home-OST0001"}
+```
+Update each target file such that the bdev path is set to the targets corresponding multipath alias. Within a few minutes IML should pick up the change on the filesystem detail page. It is important to note that IML uses longpolling to gather disk information from agent nodes so it could take up to 10 minutes for this new information to propagate up to the manager. Once it does, the target should be available to add to the filesystem and failover services should work as expected.
```
Comment (jgrund) [inline docs/support/cannot-add-osts.md]: longpolling -> polling.

Also, this is hardcoded device / fs information, which is different than the device detection IML normally does.

Are you sure that the *config store* is what's being polled? I think it's more likely that devlinks have changed (via imperative lookup in 3.x) which is why you are waiting that period. 

Also, if that is the case, it's worth mentioning that polling is a 3.x -> 4.0 construct, 4.1 should not have this wait.

Comment (johnsonw) [reply docs/support/cannot-add-osts.md]: I changed longpolling to polling. Regarding whether or not the config store is being polled, i'm not entirely sure what the details are. The devlinks change when multipath is configured, but the system didn't sync up until I updated the config store. It then took about 10 minutes before it propagated through the system such that we could transfer services from the primary to the secondary. This is a 2.4.2.7 installation and i'm guessing it's doing the same imperative lookup as 3.x.

Comment (jgrund) [reply docs/support/cannot-add-osts.md]: We're the devlinks changed during the support case?

Comment (johnsonw) [reply docs/support/cannot-add-osts.md]: From what I could tell they changed in two cases:
1. When Multipath created the alias
2. When Multipath removed the alias

I'm not sure if I'm answering your question though regarding the support case.

Comment (jgrund) [reply docs/support/cannot-add-osts.md]: I.E. did the alias (devlinks) change at all during the call?

Comment (johnsonw) [reply docs/support/cannot-add-osts.md]: Yes, during the call we checked the devlinks using udev. He then disabled Multipath and then ran udev again. This resulted in the devlinks changing to the path IML is expecting. He then turned on multipath again and we checked udev a third time which indicated that the devlinks changed to the original value when I first got on the call.

Comment (jgrund) [reply docs/support/cannot-add-osts.md]: So, yeah you are discussing two different issues here, once the item is update in the config store, you should be good to go straight away I think.

Comment (johnsonw) [reply docs/support/cannot-add-osts.md]: Ok, i'll update to reflect this.

**Comment by jgrund** [comment]
Looks good, just has a conflict then we can land.

**Pull request merged**